crates/oxc_semantic/src/builder.rs

```diff
@@ -1726,6 +1738,17 @@ impl<'a> Visit<'a> for SemanticBuilder<'a> {
             self.visit_ts_type_annotation(return_type);
         }
 
+        if expr.params.kind != FormalParameterKind::Signature && expr.params.has_parameter() {
```

Comment:
If I remember correctly, the `FormalParameterKind` of `ArrowFunctionExpression` and `Function` never going to be  `FormalParameterKind::Signature`, so if you want to call `resolve_references_for_current_scope` here, I think you don't need to check anymore.